cmd/dexctl/admin_api_connector.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/coreos/dex/client"
+	"github.com/coreos/dex/schema/adminschema"
+	"github.com/coreos/go-oidc/oidc"
+)
+
+type AdminAPIConnector struct {
+	client  *http.Client
+	baseURL string
+	apiKey  string
+}
+
+func newAdminAPIConnector(adminApiURL string, adminAPIKey string, caFile string) (*AdminAPIConnector, error) {
+	adminAPIConnector := &AdminAPIConnector{baseURL: adminApiURL}
+	var transport *http.Transport
+	if len(caFile) != 0 {
+		caCert, err := ioutil.ReadFile(caFile)
+		if err != nil {
+			return adminAPIConnector, err
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
```

Comment:
You'll want to check the return value of this call and error if it doesn't find any certs. (e.g. if someone has mistakenly used a key file)

https://golang.org/pkg/crypto/x509/#CertPool.AppendCertsFromPEM